components/pages/warning-notes/AddWarningNoteForm/AddWarningNoteForm.spec.tsx

```diff
@@ -0,0 +1,506 @@
+        month: '15',
+        year: '10',
+      });
+
+      fireEvent.submit(screen.getByRole('form'));
+
+      await waitFor(() => {
+        screen.getByText('Must be a valid Date', {
+          selector: '.govuk-error-message',
+        });
+      });
+    });
+
+    it('should show an error message if the manager discussion date entered is in the future', async () => {
+      setDateFieldValue('discussedWithManagerDate', addYears(new Date(), 1));
+
+      fireEvent.submit(screen.getByRole('form'));
+
+      await waitFor(() => {
+        screen.getByText("Date discussed with manager can't be in the future", {
+          selector: '.govuk-error-message',
+        });
+      });
+    });
+
+    it('should push to the summary route when the form is submitted with valid data', async () => {
+      setDateFieldValue('discussedWithManagerDate', addYears(new Date(), 1));
+
+      // Warning type
+      fireEvent.click(
+        screen.getByText('Risk to Self', {
+          selector: "[for='noteType_Risk to Self']",
+        })
+      );
+
+      // Start date
+      setDateFieldValue('startDate', new Date());
+
+      // Review / end date
+      setDateFieldValue('reviewDate', addMonths(new Date(), 1));
+
+      // Is the individual aware?
+      fireEvent.click(
+        screen.getByText('Yes', {
+          selector: "[for='disclosedWithIndividual_Yes']",
+        })
+      );
+
+      // Date informed
+      setDateFieldValue('disclosedDate', new Date());
+
+      // How was the individual informed?
+      fireEvent.click(
+        screen.getByText('Verbal', {
+          selector: "[for='disclosedHow_Verbal']",
+        })
+      );
+
+      // Details of disclosure to individual
+      fireEvent.change(
+        screen.getByLabelText(/Details of disclosure to individual/),
+        {
+          target: {
+            value: 'Some details',
+          },
+        }
+      );
+
+      // Warning narrative
+      fireEvent.change(
+        screen.getByLabelText(/Warning narrative and risks notes/),
+        {
+          target: {
+            value: 'Some narrative',
+          },
+        }
+      );
+
+      // Manager's name
+      fireEvent.change(screen.getByLabelText(/Manager’s name/), {
+        target: {
+          value: 'First Last',
+        },
+      });
+
+      // Date discussed with manager
+      setDateFieldValue('discussedWithManagerDate', new Date());
+
+      fireEvent.submit(screen.getByRole('form'));
+
+      await waitFor(() => {
+        expect(Router.push).toHaveBeenCalledWith(
+          '/people/100/warning-notes/add/[step]',
+          '/people/100/warning-notes/add/summary'
+        );
+      });
+    });
+
+    it('should not push to the summary route if there are errors on the form', async () => {
+      fireEvent.submit(screen.getByRole('form'));
+
+      await waitFor(() => {
+        expect(Router.push).not.toHaveBeenCalled();
+      });
+    });
+  });
+
+  describe('for a childrens resident', () => {
```

Comment:
🎉 

Wondering if we could split up the scenarios even more? 🤔 

Most of the error messages would occur on both forms. Wondering if that could be a general scenario to consider, i.e. we expect it on the form regardless of the resident, e.g. describe('when filling the form for either adults or children'), or something similar and all the form errors for the components shared between an adult and child form could go in there.

Then the form elements and associated errors specific to the adult form could go under another scenario for `describe('for an adult resident only'` or something similar.

And then a scenario for testing things on the children side, e.g. `describe('for a childrens resident only'`  or something similar.

My thoughts are around someone new/unfamiliar reading this and seeing most of the form error validations under a describe block for an adult resident and possibly misunderstand where the difference between an adult form and child form lie. E.g. Could this test file be read as the errors occur exclusively on the adults form 🤔

Reply:
Really nice suggestion, I've done that in https://github.com/LBHackney-IT/lbh-social-care-frontend/pull/464/commits/2bce97d324558a9835052ef843fdd8481ff6cca1

The test output now looks like this:

<img width="1467" alt="Screenshot 2021-06-16 at 10 56 36" src="https://user-images.githubusercontent.com/1573022/122199381-e4415a00-ce91-11eb-9de8-ab233854b724.png">